server/sonar-server/src/main/java/org/sonar/server/computation/log/LogFileRef.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.log;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import org.sonar.db.ce.CeActivityDto;
+import org.sonar.db.ce.CeQueueDto;
+import org.sonar.server.computation.CeTask;
+
+import static java.lang.String.format;
+
+public class LogFileRef {
+
+  private final String taskType;
+  private final String taskUuid;
+
+  @CheckForNull
+  private final String componentUuid;
+
+  public LogFileRef(String taskType, String taskUuid, @Nullable String componentUuid) {
+    this.taskType = taskType;
+    this.taskUuid = taskUuid;
+    this.componentUuid = componentUuid;
+  }
+
+  /**
+   * Path relative to the CE logs directory
+   */
+  public String getRelativePath() {
+    if (componentUuid == null) {
+      return format("%s/%s.log", taskType, taskUuid);
+    }
+    return format("%s/%s/%s.log", taskType, componentUuid, taskUuid);
+  }
+
+  @Override
+  public boolean equals(@Nullable Object o) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "equals" is 11 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)